lib/mutant/require_highjack.rb

```diff
@@ -10,10 +10,14 @@ module RequireHighjack
     # @return [#call]
     #   the original implementation on singleton
     def self.call(target, callback)
-      target.method(:require).tap do
+      target.public_method(:require).tap do
         target.module_eval do
+          undef_method(:require)
           define_method(:require, &callback)
-          public :require
+          class << self
```

Comment:
Should we prefer:

`singleton_class.send(:undef_method, :require)` over the `class << self` here?